wicket-extensions/src/main/java/org/apache/wicket/extensions/markup/html/form/datetime/DateTimeField.java

```diff
@@ -0,0 +1,288 @@
+	 * <p>
+	 * Note that overriding this method is a better option than overriding {@link #updateModel()}
+	 * like the first versions of this class did. The reason for that is that this method can be
+	 * used by form validators without having to depend on the actual model being updated, and this
+	 * method is called by the default implementation of {@link #updateModel()} anyway (so we don't
+	 * have to override that anymore).
+	 */
+	@Override
+	public void convertInput()
+	{
+		try
+		{
+			// Get the converted input values
+			LocalDate localDate = dateField.getConvertedInput();
+
+			if (localDate == null)
+			{
+				return;
+			}
+
+			// Use the input to create a date object with proper timezone
+			LocalTime localTime = timeField.getConvertedInput();
+			ZonedDateTime date = ZonedDateTime.of(localDate, localTime, getClientTimeZone());
+
+			// The date will be in the server's timezone
+			setConvertedInput(date);
+		}
+		catch (RuntimeException e)
+		{
+			DateTimeField.this.error(e.getMessage());
+			invalid();
+		}
+	}
+
+	/**
+	 * create a new {@link DateField} instance to be added to this panel.
+	 * 
+	 * @param id
+	 *            the component id
+	 * @param dateFieldModel
+	 *            model that should be used by the {@link DateField}
+	 * @return a new date text field instance
+	 */
+	protected DateField newDateField(String id, IModel<LocalDate> dateFieldModel)
+	{
+		return DateField.forShortStyle(id, dateFieldModel);
+	}
+
+	/**
+	 * create a new {@link TimeField} instance to be added to this panel.
+	 * 
+	 * @param id
+	 *            the component id
+	 * @param timeFieldModel
+	 *            model that should be used by the {@link TimeField}
+	 * @return a new time text field instance
+	 */
+	protected TimeField newTimeField(String id, IModel<LocalTime> timeFieldModel)
+	{
+		return TimeField.forShortStyle(id, timeFieldModel);
+	}
+
+	/**
+	 * @see org.apache.wicket.Component#onBeforeRender()
+	 */
+	@Override
+	protected void onBeforeRender()
+	{
+		dateField.setRequired(isRequired());
+		timeField.setRequired(isRequired());
+
+		ZonedDateTime modelObject = getModelObject();
+		if (modelObject == null)
+		{
+			dateTime = null;
+		}
+		else
+		{
+			// convert date to the client's time zone if we have that info
+			ZoneId zone = getClientTimeZone();
+			if (zone != null)
+			{
+				modelObject = modelObject.withZoneSameInstant(zone);
+			}
+		}
+
+		super.onBeforeRender();
+	}
+
+	protected class DateModel implements IModel<LocalDate>
+	{
+		private static final long serialVersionUID = 1L;
+
+		@Override
+		public LocalDate getObject()
+		{
+			return dateTime.toLocalDate();
+		}
+
+		@Override
+		public void setObject(LocalDate date)
+		{
+			dateTime = dateTime.with(ChronoField.YEAR, date.getYear());
+			dateTime = dateTime.with(ChronoField.MONTH_OF_YEAR, date.getMonthValue());
+			dateTime = dateTime.with(ChronoField.DAY_OF_YEAR, date.getDayOfMonth());
+		}
+
+		@Override
+		public void detach()
```

Comment:
This is not needed. The default impl in IModel does the same.

Reply:
Addressed, thanks